sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -83,6 +83,28 @@ case class SerializeFromObject(
     child: LogicalPlan) extends UnaryNode with ObjectConsumer {
 
   override def output: Seq[Attribute] = serializer.map(_.toAttribute)
+
+  // We can't estimate the size of ObjectType. We implement statistics here to avoid
+  // directly estimate any child plan which produces domain objects as output.
+  override def statistics: Statistics = {
+    if (child.output.head.dataType.isInstanceOf[ObjectType]) {
+      val underlyingPlan = child.find { p =>
```

Comment:
The ObjectProducer always sit in the middle of query plan (especially for join), the direct children of join can't be ObjectProducer.

Thinking of three operators:  SQL operator -> ObjectProducer -> ObjectConsumer (produce UnsafeRow). The data size of logical plan depends on the number of rows and the size of each row, the default size of Object only affect size of row.  The estimation of ObjectConsumer only depends on number of rows from ObjectProducer and the size of row produced by it self, this means the size of object will NOT change the size of ObjectConsumer.